src/Sylius/Bundle/CoreBundle/Tests/Functional/StateMachine/OrderCheckoutWorkflowTest.php

```diff
@@ -0,0 +1,242 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\CoreBundle\Tests\Functional\StateMachine;
+
+use PHPUnit\Framework\MockObject\MockObject;
+use Sylius\Bundle\CoreBundle\StateMachine\StateMachineInterface;
+use Sylius\Component\Core\Checker\OrderPaymentMethodSelectionRequirementCheckerInterface;
+use Sylius\Component\Core\Checker\OrderShippingMethodSelectionRequirementCheckerInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\Customer;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\OrderShippingStates;
+use Sylius\Component\Core\Repository\PromotionRepositoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class OrderCheckoutWorkflowTest extends KernelTestCase
+{
+    /** @var OrderShippingMethodSelectionRequirementCheckerInterface|MockObject */
+    private $orderShippingMethodSelectionRequirementChecker;
+
+    /** @var OrderPaymentMethodSelectionRequirementCheckerInterface|MockObject */
+    private $orderPaymentMethodSelectionRequirementChecker;
+
+    public function setup(): void
+    {
+        parent::setUp();
+        $promotionRepository = $this->createMock(PromotionRepositoryInterface::class);
```

Comment:
I assume promotion and sequence repositories are needed as there is no direct usage of them